backend/test/unit/TrialStateChangeEvents.unit.test.js

```diff
@@ -0,0 +1,125 @@
+        const playerB = new Player(null, null, 'b', roles.JESTER, true);
+        const playerC = new Player(null, null, 'c', roles.CIVILIAN, true);
+
+        MafiaGameMock.addPlayer(playerA, roomElements.roomID);
+        MafiaGameMock.addPlayer(playerB, roomElements.roomID);
+        MafiaGameMock.addPlayer(playerC, roomElements.roomID);
+
+        clientSocket.on('trial-start', (trialStartDTO) => {
+            expect(trialStartDTO.timeToVote).toBe(config.trial_total_vote_time_in_milliseconds);
+        });
+
+        clientSocket.on('trial-end', (trialEndDTO) => {
+            expect(trialEndDTO.playerKilled).toBe('abstain Vote');
+            expect(trialEndDTO.isGameOver).toBe(false);
+            done();
+        });
+
+        clientSocket.emit('start-trial');
+    });
+
+    test('trial-start successful call, one abstain vote, winning condition is fulfilled', (done) => {
+        const playerA = new Player(null, null, 'a', roles.MAFIA, true);
+        playerA.isAlive = false; // Kill off the mafia player
+
+        MafiaGameMock.addPlayer(playerA, roomElements.roomID);
+
+        MafiaGameMock.addTrialVote(playerA, 'abstain Vote', roomElements.roomID); // Vote to kill off a civilian
+
+        clientSocket.on('trial-start', (trialStartDTO) => {
+            expect(trialStartDTO.timeToVote).toBe(config.trial_total_vote_time_in_milliseconds);
+        });
+
+        clientSocket.on('trial-end', (trialEndDTO) => {
+            expect(trialEndDTO.playerKilled).toBe('abstain Vote');
+            expect(trialEndDTO.isGameOver).toBe(true);
+            done();
+        });
+
+        clientSocket.emit('start-trial');
+    });
+
+    test('trial-start successful call, someone is killed, winning condition is not fulfilled', (done) => {
+        const playerA = new Player(null, null, 'a', roles.MAFIA, true);
+        const playerB = new Player(null, null, 'b', roles.CIVILIAN, true);
+        const playerC = new Player(null, null, 'c', roles.JESTER, true);
+        const playerD = new Player(null, null, 'd', roles.CIVILIAN, true);
+
+        MafiaGameMock.addPlayer(playerA, roomElements.roomID);
+        MafiaGameMock.addPlayer(playerB, roomElements.roomID);
+        MafiaGameMock.addPlayer(playerC, roomElements.roomID);
+        MafiaGameMock.addPlayer(playerD, roomElements.roomID);
+
+        MafiaGameMock.addTrialVote(playerA, playerD, roomElements.roomID); // Vote to kill off a civilian
+
+        clientSocket.on('trial-start', (trialStartDTO) => {
+            expect(trialStartDTO.timeToVote).toBe(config.trial_total_vote_time_in_milliseconds);
+        });
+
+        clientSocket.on('trial-end', (trialEndDTO) => {
+            expect(trialEndDTO.playerKilled).toBe('d');
+            expect(trialEndDTO.isGameOver).toBe(false);
+            done();
+        });
+
+        clientSocket.emit('start-trial');
+    });
+
+    test('trial-start successful call, someone is killed, winning condition is fulfilled', (done) => {
```

Comment:
I think a more realistic test case would be to add another Civilian into the game at this point. So that we enter the trial with three players! I'm not sure if our current flow will ever allow one mafia and one civilian to enter a trial.

Reply:
Yup, that sounds reasonable. I changed it to fit what you suggested :)